### DISCUSSION TITLE
windows old system

### DISCUSSION THREAD

**Discussion by xiaokaidada** [zauberzeug/nicegui#1926] [Q&A] [answered]
### Question

Nicegui doesn't support running under windows in the old system, my system version is Windows Server 2012 R2, I can't open the exe program, but when I open it, I see the program appear in the task manager, it seems to self-destruct, should I install the environment on the old system?

Here's the code I tested
`from nicegui import ui

ui.label('demo')
ui.button('demo')

ui.run(native=True, reload=False)`

**Comment by xiaokaidada** [comment]
Some servers work, but the page is blank

**Reply by xiaokaidada** [reply]
It seems that most servers do not support nicegui, some can't open, some open the page blank, but install node.js can display the page, maybe some servers are too old?

**Comment by xiaokaidada** [comment]
![photo_2023-10-30_19-28-46](https://github.com/zauberzeug/nicegui/assets/32946616/8f530031-9cd8-4faf-99b0-a00cf333a77f)

**Comment by rodja** [comment]
>  I can't open the exe program,

Sorry, I do not understand. What exe program? Do you use pyinstaller?

**Reply by xiaokaidada** [reply]
Yes, I used pyinstaller to package it, but the program is on a blank page after packaging and running, do some older servers need to install any specific environment to run nicegui perfectly?

**Reply by rodja** [reply]
I think that pyinstaller creates self-contained executables. But there could be a bug with pyinstaller, pywebview or NiceGUI. Have you followed the explanations https://nicegui.io/documentation#package_for_installation? And what happens when you use "native=False"?

**Reply by xiaokaidada** [reply]
You see I've reposted a comment

**Comment by xiaokaidada** [comment]
![GIF 2023-10-31 16-02-04](https://github.com/zauberzeug/nicegui/assets/32946616/65672ad8-fa26-4ba9-ae9e-10e3b2abe4ad)
"native=False" is normal, it will launch the browser and display the page, but when "native=True" appears in the screen, it is not clear why it will automatically destroy when it is launched

**Reply by xiaokaidada** [reply]
Corresponding computer configuration
![微信截图_20231031161009](https://github.com/zauberzeug/nicegui/assets/32946616/8be10d9d-90f1-4a7e-8acb-243eb4035745)

**Comment by rodja** [comment]
Could you please try to package and run a simple pywebview application? That way we could find out if its NiceGUI related...

**Reply by xiaokaidada** [reply, marked as answer]
It seems to be a problem with pywebview, not nicegui, pywebview is not compatible enough
In pywebview, I tried the following code with the exact same symptoms as the gif
import webview
webview.create_window('Hello world', 'https://pywebview.flowrl.com/hello')
webview.start()

**Reply by rodja** [reply]
Ok, I'll close the discussion for now. Feel free to reopen if something is off with NiceGUI or post a link to the pywebview issue if one exists.

**Reply by xiaokaidada** [reply]
Ok thank you
